security/s2a/internal/crypter/aesgcm_test.go

```diff
@@ -0,0 +1,474 @@
+			AllocateDst: true,
+		},
+		{
+			Key:         testutil.Dehex("11754cd72aec309bf52f7687212e8957"),
+			Plaintext:   nil,
+			Ciphertext:  nil,
+			Tag:         testutil.Dehex("250327c674aaf477aef2675748cf6971"),
+			Nonce:       testutil.Dehex("3c819d9a9bed087615030b65"),
+			Result:      testutil.ValidResult,
+			AllocateDst: false,
+		},
+		{
+			Key:         testutil.Dehex("ca47248ac0b6f8372a97ac43508308ed"),
+			Plaintext:   nil,
+			Ciphertext:  nil,
+			Tag:         testutil.Dehex("60d20404af527d248d893ae495707d1a"),
+			Nonce:       testutil.Dehex("ffd2b598feabc9019262d2be"),
+			Result:      testutil.ValidResult,
+			AllocateDst: false,
+		},
+		{
+			Key:         testutil.Dehex("7fddb57453c241d03efbed3ac44e371c"),
+			Plaintext:   testutil.Dehex("d5de42b461646c255c87bd2962d3b9a2"),
+			Ciphertext:  testutil.Dehex("2ccda4a5415cb91e135c2a0f78c9b2fd"),
+			Tag:         testutil.Dehex("b36d1df9b9d5e596f83e8b7f52971cb3"),
+			Nonce:       testutil.Dehex("ee283a3fc75575e33efd4887"),
+			Result:      testutil.ValidResult,
+			AllocateDst: false,
+		},
+		{
+			Key:         testutil.Dehex("ab72c77b97cb5fe9a382d9fe81ffdbed"),
+			Plaintext:   testutil.Dehex("007c5e5b3e59df24a7c355584fc1518d"),
+			Ciphertext:  testutil.Dehex("0e1bde206a07a9c2c1b65300f8c64997"),
+			Tag:         testutil.Dehex("2b4401346697138c7a4891ee59867d0c"),
+			Nonce:       testutil.Dehex("54cc7dc2c37ec006bcc6d1da"),
+			Result:      testutil.ValidResult,
+			AllocateDst: false,
+		},
+		{
+			Key:         testutil.Dehex("11754cd72aec309bf52f7687212e8957"),
+			Plaintext:   nil,
+			Ciphertext:  nil,
+			Tag:         testutil.Dehex("250327c674aaf477aef2675748cf6971"),
+			Nonce:       testutil.Dehex("3c819d9a9bed087615030b65"),
+			Result:      testutil.ValidResult,
+			AllocateDst: true,
+		},
+		{
+			Key:         testutil.Dehex("ca47248ac0b6f8372a97ac43508308ed"),
+			Plaintext:   nil,
+			Ciphertext:  nil,
+			Tag:         testutil.Dehex("60d20404af527d248d893ae495707d1a"),
+			Nonce:       testutil.Dehex("ffd2b598feabc9019262d2be"),
+			Result:      testutil.ValidResult,
+			AllocateDst: true,
+		},
+		{
+			Key:         testutil.Dehex("7fddb57453c241d03efbed3ac44e371c"),
+			Plaintext:   testutil.Dehex("d5de42b461646c255c87bd2962d3b9a2"),
+			Ciphertext:  testutil.Dehex("2ccda4a5415cb91e135c2a0f78c9b2fd"),
+			Tag:         testutil.Dehex("b36d1df9b9d5e596f83e8b7f52971cb3"),
+			Nonce:       testutil.Dehex("ee283a3fc75575e33efd4887"),
+			Result:      testutil.ValidResult,
+			AllocateDst: true,
+		},
+		{
+			Key:         testutil.Dehex("ab72c77b97cb5fe9a382d9fe81ffdbed"),
+			Plaintext:   testutil.Dehex("007c5e5b3e59df24a7c355584fc1518d"),
+			Ciphertext:  testutil.Dehex("0e1bde206a07a9c2c1b65300f8c64997"),
+			Tag:         testutil.Dehex("2b4401346697138c7a4891ee59867d0c"),
+			Nonce:       testutil.Dehex("54cc7dc2c37ec006bcc6d1da"),
+			Result:      testutil.ValidResult,
+			AllocateDst: true,
+		},
+	} {
+		sender, receiver := getGCMCryptoPair(test.Key, t)
```

Comment:
Should we use `t.Run` here as well?

Reply:
Yes, with the addition of `Comment`, I think that makes sense.